Source/WebCore/bindings/js/JSFileSystemWritableFileStreamCustom.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "JSFileSystemWritableFileStream.h"
+
+#include "InternalWritableStream.h"
+#include "JSDOMPromiseDeferred.h"
+#include "WebCoreJSClientData.h"
+#include <JavaScriptCore/CallFrame.h>
+
+namespace WebCore {
+
+static JSC::JSValue writeImpl(FileSystemWritableFileStream& stream, JSC::JSGlobalObject& lexicalGlobalObject, JSC::JSValue data)
+{
+    auto& vm = JSC::getVM(&lexicalGlobalObject);
+    auto* clientData = static_cast<JSVMClientData*>(vm.clientData);
+
+    JSC::MarkedArgumentBuffer arguments;
+    arguments.append(stream.internalWritableStream().guardedObject());
+    ASSERT(!arguments.hasOverflowed());
+    auto& writerPrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().acquireWritableStreamDefaultWriterPrivateName();
+    auto writerOrException = InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, writerPrivateName, arguments);
```

Comment:
Could we instead move that code to InternalWritableStream instead as a helper routine?

Reply:
Moved to `InternalWritableStream`